projects/packages/waf/src/class-compatibility.php

```diff
@@ -81,4 +96,122 @@ public static function default_option_waf_needs_update( $default, $option, $pass
 		return true;
 	}
 
+	/**
+	 * Merge the WAF and Brute Force Protection IP allow lists.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $waf_allow_list        The WAF IP allow list.
+	 * @param array  $brute_force_allow_list The Brute Force Protection IP allow list. Array of IP objects.
+	 *
+	 * @return string The merged IP allow list.
+	 */
+	public static function merge_ip_allow_lists( $waf_allow_list, $brute_force_allow_list ) {
+
+		if ( empty( $brute_force_allow_list ) ) {
+			return $waf_allow_list;
+		}
+
+		// Convert the IP objects to strings.
+		$brute_force_allow_list = array_map(
+			function ( $ip_object ) {
+				if ( isset( $ip_object->range ) && $ip_object->range ) {
```

Comment:
OTOH, this would be a good place to use `empty()`, as `isset( $x ) && $x` is exactly what `! empty( $x )` does.
```suggestion
				if ( ! empty( $ip_object->range ) ) {
```
Unless, of course, the `isset()` call isn't actually needed here. I see you're accessing other elements of the object without any checks.

Reply:
Thanks for these notes! 😁

This `isset()` is actually needed, as some IP objects don't include the range.

Maybe A bit loosey-goosey, but if the `range` property is set, then the `range_low` and `range_high` should be available. Similarly, if `range` property is _not_ set, the `ip_address` property will be available. So I'm only bothering to check if `range` exists.